pkg/schedulingtypes/plugin.go

```diff
@@ -161,59 +126,45 @@ func (p *Plugin) TemplateExists(key string) bool {
 	return exist
 }
 
-func (p *Plugin) ReconcilePlacement(qualifiedName util.QualifiedName, newClusterNames []string) error {
-	placement, err := p.placementClient.Resources(qualifiedName.Namespace).Get(qualifiedName.Name, metav1.GetOptions{})
-	if err != nil {
-		if !apierrors.IsNotFound(err) {
-			return err
-		}
-		newPlacement := newUnstructured(p.typeConfig.GetPlacement(), qualifiedName)
-		setPlacementSpec(newPlacement, newClusterNames)
-		_, err := p.placementClient.Resources(qualifiedName.Namespace).Create(newPlacement, metav1.CreateOptions{})
-		return err
+func (p *Plugin) Reconcile(qualifiedName util.QualifiedName, result map[string]int64) error {
+	fedObject, err := p.federatedTypeClient.Resources(qualifiedName.Namespace).Get(qualifiedName.Name, metav1.GetOptions{})
+	if err != nil && apierrors.IsNotFound(err) {
+		// Federated resource has been deleted - no further action required
+		return nil
 	}
-
-	clusterNames, err := util.GetClusterNames(placement)
 	if err != nil {
 		return err
 	}
-	if PlacementUpdateNeeded(clusterNames, newClusterNames) {
-		setPlacementSpec(placement, newClusterNames)
-		_, err := p.placementClient.Resources(qualifiedName.Namespace).Update(placement, metav1.UpdateOptions{})
-		if err != nil {
-			return err
-		}
-	}
 
-	return nil
-}
+	isDirty := false
 
-func (p *Plugin) ReconcileOverride(qualifiedName util.QualifiedName, result map[string]int64) error {
-	override, err := p.overrideClient.Resources(qualifiedName.Namespace).Get(qualifiedName.Name, metav1.GetOptions{})
+	newClusterNames := []string{}
+	for name := range result {
+		newClusterNames = append(newClusterNames, name)
+	}
+	clusterNames, err := util.GetClusterNames(fedObject)
```

Comment:
`util.GetClusterNames` seems only work if placement contains slice of clusterNames and does not work with label based clusterSelector. Better to add a TODO, if we could not solve it in this pr, which could be taken up later.

Reply:
`GetClusterNames` considers both `clusterNames` and `clusterSelector`

Reply:
@shashidharatd, if an RSP is present, it will always override the `placement.clusternames` to the names which result from scheduling the per cluster user preferences specified in the RSP. It will write new `placement.clusternames` if not already present on the object. `GetClusterNames()` is used here to get and determine if a write() on the object is needed or not, and avoid write() if the object already has the same `clusternames`.

Reply:
ok. SGTM for now